tests/ui/async-await/in-trait/warn.stderr

```diff
@@ -0,0 +1,20 @@
+error: use of `async fn` in public traits is discouraged as auto trait bounds cannot be specified
+  --> $DIR/warn.rs:7:5
+   |
+LL |     async fn not_send();
+   |     ^^^^^
+   |
+   = note: you can suppress this lint if you plan to use the trait locally, for concrete types, or do not care about auto traits like `Send` on the `Future`
```

Comment:
```suggestion
   = note: you can suppress this lint if you plan to use the trait only in your own code, or do not care about auto traits like `Send` on the `Future`
```

I think the emphasis on concrete types is confusing. I can't think of an example of a trait a user would write that is never used in generics.

Reply:
Where this would come up is the same places that someone would want to use inherent trait impls.

Reply:
But overall I agree the language is a bit fussy, so +1 on the change.